libs/functions/calculate_projections.py

```diff
@@ -0,0 +1,99 @@
+import pandas as pd
+import datetime
+import os.path
+import simplejson
+
+from libs.us_state_abbrev import us_state_abbrev
+from libs.datasets import FIPSPopulation
+from libs.datasets.model_output_schema import MODEL_OUTPUT_SCHEMA, MODEL_OUTPUT_SCHEMA_EXCLUDED_COLUMNS
+from libs.datasets.projections_schema import CALCULATED_PROJECTION_HEADERS_STATES, CALCULATED_PROJECTION_HEADERS_COUNTIES
+
+def _calc_short_fall(x):
+    return abs(x.beds - x.all_hospitalized) if x.all_hospitalized > x.beds else 0
+
+def _get_hospitals_and_shortfalls(df, date_out):
+    first_record_after_date = df[(df.date > date_out)].iloc[0]
+    hospital_short_fall_columns = ['all_hospitalized', 'short_fall']
+    return tuple(first_record_after_date[hospital_short_fall_columns].values)
+
+def _read_json_as_df(path):
+    # TODO: read this from a dataset class
+    df = pd.DataFrame.from_records(simplejson.load(open(path,'r')),
+    columns=MODEL_OUTPUT_SCHEMA, exclude=MODEL_OUTPUT_SCHEMA_EXCLUDED_COLUMNS)
```

Comment:
Just thinking here - I think this could be it's own `DataSource` that just has columns defined (like `JHUDataset`)

(definitely not for this pr lol)

Reply:
ya i think ideally we would use datasource for this that have the headers named and defined in the class so we wouldn't have lists of strings